src/folio_migration_tools/migration_tasks/batch_poster.py

```diff
@@ -581,8 +591,8 @@ def post_batch(self, batch, failed_recs_file, num_records, recursion_depth=0):
                 resp = json.dumps(response, indent=4)
             except TypeError:
                 resp = response
-            except Exception:
-                logging.exception("something unexpected happened")
+            except (ValueError, RecursionError) as e:
```

Comment:
Curious about the change here. I know that using the base exception here isn't generally a best-practice, but we want to make sure we're catching any possible exception here so we can raise the record failed error.